processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -469,7 +469,8 @@ public boolean isAssignableTo(Type other) {
             // the current target accessor can also be a getter method.
             // The following if block, checks if the target accessor should be overruled by an add method.
             if ( cmStrategy == CollectionMappingStrategyPrism.SETTER_PREFERRED
-                || cmStrategy == CollectionMappingStrategyPrism.ADDER_PREFERRED ) {
+                || cmStrategy == CollectionMappingStrategyPrism.ADDER_PREFERRED
+                || cmStrategy == CollectionMappingStrategyPrism.TARGET_IMMUTABLE ) {
```

Comment:
The check is added, but inside property mapping.. 

Looking at this code, I'm thinking.. Is the table correct in the documentation? The documentation has a column "Only target add- Available".. That is not possible. The adder is derived from the presence of either a setter and/or a getter. Note: candidates only contains getters/setters... WDYT?

If this is wrong.. we should handle it in a separate PR..